apps/jsonrpc2/lib/jsonrpc2/response/helpers.ex

```diff
@@ -1,19 +1,40 @@
 defmodule JSONRPC2.Response.Helpers do
-  @spec encode_hex(binary() | integer() | nil) :: binary()
-  def encode_hex(binary) when is_binary(binary) do
-    Exth.encode_hex(binary)
+  @moduledoc """
+  Encodes an usigned integer as a binary as described in Ethereum JSON RPC spec.
+  """
+  @spec encode_quantity(binary() | non_neg_integer() | nil) :: binary() | nil
+  def encode_quantity(binary) when is_binary(binary) do
+    hex_binary = Base.encode16(binary, case: :lower)
+
+    case hex_binary do
```

Comment:
I don't think you need to check this case as it should be covered by the check on line 16

Reply:
I think removing this case would change the logic. I'm not sure if it _should_ change, but I think it would change. For instance, if this is hit, and `hex_binary` is `""`, `""` will be returned. If removed, `"0x0"` will be returned.

Reply:
@willmeister yep, good concern. but `hex_binary` shouldn't be `""`, because only encoded integers are passed to `encode_quantity` function